once_test.py

```diff
@@ -415,6 +416,22 @@ def test_threaded_single_function(self):
             self.assertEqual(r, 1)
         self.assertEqual(counter.value, 1)
 
+    def test_once_per_thread(self):
+        counter = Counter()
+
+        @execute_with_barrier(n_workers=_N_WORKERS)
+        @once.once(per_thread=True)
```

Comment:
I think you want to flip these so that the barrier is inside the once, not outside

Reply:
Made it into a sandwich!

Reply:
Why the sandwich? Just the inside one should be needed!